tools/fetch-options-plugin.ts

```diff
@@ -0,0 +1,71 @@
+import { WarpPlugin, WarpPluginType } from '../src/core/WarpPlugin';
+import { FetchRequest } from '../src/core/WarpFetchWrapper';
+import { JWKInterface } from 'arweave/node/lib/wallet';
+import fs from 'fs';
+import path from 'path';
+import { LoggerFactory } from '../src/logging/LoggerFactory';
+import { defaultCacheOptions, WarpFactory } from '../src/core/WarpFactory';
+
+class FetchOptionsPlugin implements WarpPlugin<FetchRequest, RequestInit> {
+  process(request: FetchRequest): Partial<RequestInit> {
+    const url = request.input;
+
+    let fetchOptions = request.init;
+
+    if (url == `https://d1o5nlqr4okus2.cloudfront.net/gateway/sequencer/register`) {
+      fetchOptions = {
```

Comment:
hmm, maybe we should merge the options in the wrapper itself? Imho there's a risk that a plugin developer may forget to do a manual merge...

Reply:
we can, it seemed easier for me to just update the object but maybe you're right.

Reply:
I believe it's more important to make it easier for developers to use than for us to implement ;-)